### PR TITLE
TNM-4076 Add ability for a party to unregister

### PR DESCRIPTION
**Pull request by DStranger**

### CODE REVIEW

File: endpoints-registration/src/main/scala/com/thenewmotion/ocpi/registration/RegistrationService.scala [hunk 1]
```diff
@@ -66,7 +70,7 @@ class RegistrationService(
       ).map(_.right))
     } yield generateCreds(newTokenToConnectToUs)).run.map {
       _.leftMap {
-        e => logger.error(s"error during reactToPostCredsRequest: $e"); e
+        e => logger.error("error during reactToPostCredsRequest", e); e
```
Comment (kiequoo) [inline endpoints-registration/src/main/scala/com/thenewmotion/ocpi/registration/RegistrationService.scala]: `e` in this case is not `Throwable`, so don't you also need `{}` in the string?

File: msgs/src/main/scala/com/thenewmotion/ocpi/msgs/OcpiStatusCode.scala [hunk 2]
```diff
@@ -41,6 +41,7 @@ object OcpiStatusCode {
   val PartyAlreadyRegistered = ClientErrorCode(2012) // When POSTing
   val RegistrationNotCompletedYetByParty = ClientErrorCode(2013) // When PUTing or GETing even
   val AuthorizationFailed = ClientErrorCode(2014)
+  val ClientWasNotRegistered = ClientErrorCode(2015)
```
Comment (kiequoo) [inline msgs/src/main/scala/com/thenewmotion/ocpi/msgs/OcpiStatusCode.scala]: Was this in the spec somewhere or you just picked the next number?  I'm not sure where these other error codes came from TBH

File: endpoints-registration/src/main/scala/com/thenewmotion/ocpi/registration/RegistrationService.scala (hunk 1 above)
Comment (DStranger) [inline endpoints-registration/src/main/scala/com/thenewmotion/ocpi/registration/RegistrationService.scala]: True, I thought it was an exception...

File: msgs/src/main/scala/com/thenewmotion/ocpi/msgs/OcpiStatusCode.scala (hunk 2 above)
Comment (DStranger) [inline msgs/src/main/scala/com/thenewmotion/ocpi/msgs/OcpiStatusCode.scala]: Yes, those starting from 2010 were created by us.

**Pull request merged**